Sources/MapboxMaps/Foundation/Extensions/Core/MapEvents+Deprecated.swift

```diff
@@ -0,0 +1,270 @@
+import Foundation
+
+public extension MapEvents {
+    /// Events emitted by the SDK
+    @available(*, deprecated, message: "Use 'Event' instead")
+    enum EventKind: RawRepresentable, CaseIterable {
```

Comment:
Does it generate a deprecation warning in the type declaration line of code OR in the line we use the concrete case (like `.mapLoaded`)?